server/container_create_linux.go

```diff
@@ -1076,18 +1076,32 @@ func setupSystemd(mounts []rspec.Mount, g generate.Generator) {
 		}
 		g.AddMount(systemdMnt)
 	} else {
-		systemdMnt := rspec.Mount{
-			Destination: "/sys/fs/cgroup/systemd",
-			Type:        "bind",
-			Source:      "/sys/fs/cgroup/systemd",
-			Options:     []string{"bind", "nodev", "noexec", "nosuid"},
+		// If the /sys/fs/cgroup is bind mounted from the host,
+		// then systemd-mode cgroup should be disabled
+		// https://bugzilla.redhat.com/show_bug.cgi?id=2064741
+		if doesntHaveSysFsCgroupFromHost(g.Mounts()) {
+			systemdMnt := rspec.Mount{
```

Comment:
what happens when we just don't add this mount? in addOCIBindMounts, we add `/sys/fs/cgroup` if we haven't already mounted `/sys`. This means, theoretically, this code block would only run if we're mounting sys, which would inherit `/sys/fs/cgroup/systemd` anyway, right? (cc @giuseppe )

Reply:
@haircommander You are right! Completely removing that code block also fixes the issue. 

https://github.com/cri-o/cri-o/pull/5785/files#diff-1e0877112035840d3e2f073c693665f8daffa241a440e54a866ad7382b510ea2R1079-R1093

We are handling this already in `addOCIBindMounts` function for both cgroups v1 and v2,

https://github.com/cri-o/cri-o/blob/main/server/container_create_linux.go#L1010-L1024

So IMHO, we can completely get rid of,

https://github.com/cri-o/cri-o/blob/main/server/container_create_linux.go#L1069-L1085

@giuseppe @haircommander WDYT?

Reply:
cc @saschagrunert

Reply:
less code is better to me!

Reply:
I am not sure that would work.

on cgroup v2, systemd requires `/sys/fs/cgroup` to be writeable, while on cgroupv1 it requires `/sys/fs/cgroup/systemd` to be writeable.

The mount could exist when we bind mount `/sys/fs/cgroup` but it is not writeable

Reply:
I haven't tested in cgroups v2, but I guess for now we can leave the cgroups v2 part as is and only update the code related to cgroups v1

Reply:
The existing code already takes care of mounting with appropriate read/write options, 

https://github.com/cri-o/cri-o/blob/main/server/container_create_linux.go#L1018-L1023

Reply:
the existing code, AFAICS, depends on an annotation to be present, and it doesn't address `/sys/fs/cgroup/systemd` being writeable on cgroup v1